src/Symfony/Component/Validator/Constraints/UniqueValidator.php

```diff
@@ -43,7 +43,7 @@ public function validate(mixed $value, Constraint $constraint)
         $collectionElements = [];
         $normalizer = $this->getNormalizer($constraint);
         foreach ($value as $element) {
-            if ($fields && !$element = $this->reduceElementKeys($fields, $element)) {
+            if ($fields && !(\is_array($element) && $element = $this->reduceElementKeys($fields, $element))) {
```

Comment:
I wonder if we shouldn't rather throw an `UnexpectedValueException` instead.

Reply:
Good point!

Checking types is not this constraint validator responsibility, as it doesn't need a certain type to do its validation.

When `fields` is passed, and an item doesn't have them, even if it's an array, it is ignored. Why not do the same when the item is not an array?

Every item in the following collection is unique, event when passing `fields: 'a'` to the constraint:
```php
[
  ['a' => 1, 'b' => 1],
  ['a' => 2, 'b' => 2],
  ['b' => 3],
  null,
]
```

Btw, I'll be happy with any solution, as long as it doesn't crash.

---

On a wider thought, I think the `UnexpectedValueException` should not exist, as we already have the `Type` constraint.
A value that can not be validated by a constraint should be ignored (ie: `EmailValidator` should skip anything that's not a string)
But this is another topic, and probably I lack the knowledge to suggest so 😅